### ISSUE TITLE
Hanabi: Obs & Action Space, Config Parameters, and Updating Docs/Docstrings

### ISSUE THREAD

**Issue by jdpena** [PettingZoo-Team/PettingZoo#235]
Hey fellas, 

**TLDR**; does PettingZoo's `hanabi_v0` have the exact same observation and action space as DeepMind's (DMs) [hanabi-learning-environment](https://github.com/deepmind/hanabi-learning-environment) (HLE), and does it properly handle config parameters? 

I was trying to run some hanabi experiments and I just wanted to confirm a few things since I unfortunately haven't been able to find any thorough documentation on the observation and action space of DM's HLE.

I see PettingZoo's `hanabi_v0` env wraps around DM's HLE. When I run the `hanabi_v0` environment, however, I get an observation space size of 373 (as noted in your [hanabi docs](https://www.pettingzoo.ml/classic/hanabi)), and when I run DM's [hanabi-learning-environment/examples/game_example.py](https://github.com/deepmind/hanabi-learning-environment/blob/master/examples/game_example.py), I get an observation space of size 658.

I dug around both codebases to find the differences and it looks like it's all due to the initial setup of the game. `hanabi_v0` initializes the `hand_size` parameter to 2, whereas the default parameter in HLE is 5 ([hanabi_lib/hanabi_game.cc#L147](https://github.com/deepmind/hanabi-learning-environment/blob/master/hanabi_learning_environment/hanabi_lib/hanabi_game.cc#L147)). If I pass `{"hand_size": 2}` as the argument in [hanabi-learning-environment/examples/game_example.py#L113](https://github.com/deepmind/hanabi-learning-environment/blob/master/examples/game_example.py#L113), I obtain a matching observation space of size 373.

I would like to set `hand_size=5` in `hanbi_v0`, but I don't see the parameters listed as options in PettingZoo's [hanabi docs](https://www.pettingzoo.ml/classic/hanabi). Does `hanabi_v0` allow for the additional parameters? The function signature and docstrings in [classic/hanabi/hanabi.py](https://github.com/PettingZoo-Team/PettingZoo/blob/master/pettingzoo/classic/hanabi/hanabi.py) ([hanabi.py#L54](https://github.com/PettingZoo-Team/PettingZoo/blob/master/pettingzoo/classic/hanabi/hanabi.py#L54) and [hanabi.py#L60](https://github.com/PettingZoo-Team/PettingZoo/blob/master/pettingzoo/classic/hanabi/hanabi.py#L60), respectively) suggest it is possible to pass the arguments. I'm just not sure if this will break something under-the-hood.

Moreover, the docstrings in `hanabi.py` seem slightly incomplete and/or incorrect. For example, the "Common game configurations" do not match their HLE counterparts, e.g., "Hanabi-Small" in `hanabi_v0`'s docstring in [classic/hanabi/hanabi.py#92](https://github.com/PettingZoo-Team/PettingZoo/blob/master/pettingzoo/classic/hanabi/hanabi.py#L92) vs. HLE's actual config in [hanabi_learning_environment/rl_env.py#L552](https://github.com/deepmind/hanabi-learning-environment/blob/master/hanabi_learning_environment/rl_env.py#L552), among others.

But I digress.. 

I can't find any thorough documentation on HLE's observation and action space -- the PettingZoo [hanabi docs](https://www.pettingzoo.ml/classic/hanabi) are the most complete I have found. 

I have looked through the DM's HLE code that defines the observation space ([hanabi_lib/hanabi_observation.h](https://github.com/deepmind/hanabi-learning-environment/blob/master/hanabi_learning_environment/hanabi_lib/hanabi_observation.h), [hanabi_lib/hanabi_observation.cc](https://github.com/deepmind/hanabi-learning-environment/blob/master/hanabi_learning_environment/hanabi_lib/hanabi_observation.cc), and [hanabi_lib/canonical_encoders.cc](https://github.com/deepmind/hanabi-learning-environment/blob/master/hanabi_learning_environment/hanabi_lib/canonical_encoders.cc)) and it seems very similar to what's defined in the [hanabi docs](https://www.pettingzoo.ml/classic/hanabi). 

I don't see any special/additional transformations of the observations in `hanabi_v0`, so I just want to make sure they are equivalent and ordered the same, even when changing `hanabi_v0`'s configs (e.g., `hand_size`).

Thanks,

Jaime

**Comment by weepingwillowben** [comment]
Hi Jamie,

If you have a particular configuration in mind, you can set the parameters here:
https://github.com/PettingZoo-Team/PettingZoo/blob/master/pettingzoo/classic/hanabi/hanabi.py#L54

I guess they are not documented in the main documentation (whoops!), but they should all work correctly. 

As for the default Large/Small/very small setups, I'll make sure to fix those. There are some weird complications though. I have found that when the number of players is 5, hanabi crashes unless the hand size is 4 (as specified here https://github.com/deepmind/hanabi-learning-environment/blob/master/hanabi_learning_environment/hanabi_lib/hanabi_game.cc#L147). I guess these points should also be added to the documentation.

**Comment by weepingwillowben** [comment]
Edit, it seems as though the crash happens with combinations of `colors` and `players`, not hand size. Anyways, the point is that certain combinations don't really work.

**Comment by weepingwillowben** [comment]
As for observations, we just pass through whatever we get from the hanabi_learning_library without any changes. I did not write the documentation, but I believe it was created by going though and looking at the code very much like what you are doing.

**Comment by jdpena** [comment]
Thanks for your response Ben. I see the docstrings changes were merged and I'm sure the web API will be updated soon, so I can close out this issue.

I haven't run into crashing issues with the hanabi env, but it appears to be DM abandoned their hanabi-learning-environment repo (the last meaningful updates/changes occurred about a year ago). FAIR forked their project and has a stable version of it, where they added some new functionality and a made few fixes ([hengyuan-hu/hanabi-learning-environment](https://github.com/hengyuan-hu/hanabi-learning-environment/tree/stable)). They're maintaining it for their ["Other-Play" for Zero-Shot Coordination](https://arxiv.org/abs/2003.02979) and [Simplified Action Decoder for Deep Multi-Agent Reinforcement Learning](https://arxiv.org/abs/1912.02288) research. 

I know the PettingZoo team maintains the pypi version of the env, so have you guys thought of maybe switching to the updated FAIR env? The main changes are listed here [hanabi-learning-environment/compare/master...hengyuan-hu:273c5ca7f5](https://github.com/deepmind/hanabi-learning-environment/compare/master...hengyuan-hu:273c5ca7f5)

**Comment by justinkterry** [comment]
We'll try to fix the documentation this week, sorry it's taken so long. 

Would you be interested in doing a PR to upstream the FAIR fixes to us? We're pretty thoroughly overwhelmed with the list of outstanding fixes.

**Comment by justinkterry** [comment]
So #251 fixed the key issues here, thanks a ton for bringing this to our attention

**Comment by jdpena** [comment]
> We'll try to fix the documentation this week, sorry it's taken so long.
> 
> Would you be interested in doing a PR to upstream the FAIR fixes to us? We're pretty thoroughly overwhelmed with the list of outstanding fixes.

I forgot to mention that I'm currently looking into the FAIR hanabi env changes, but it's been a slow process. I'll issue a PR to your team's hanabi-learning-environment repo once I'm able to get it all sorted it out.

Thanks for all your help with this!
